src/test/java/uk/ac/ebi/embl/template/TemplateEntryProcessorTest.java

```diff
@@ -0,0 +1,170 @@
+    }
+    
+    @Test
+    public void  testEntryProcess_ERT000002() throws Exception {
+
+        TemplateInfo templateInfo_ERT000002 = new TemplateLoader().loadTemplateFromFile(templateFile);
+        String MOL_TYPE_ERT000002 = getMolTypeFromTemplateForTest(templateInfo_ERT000002);
+
+        // Test with biosample Id
+        executeEntryProcessTestWithSample("SAMEA9403245",templateInfo_ERT000002 ,MOL_TYPE_ERT000002 );
+        
+        // Test with sample Id
+        executeEntryProcessTestWithSample("ERS7118926",templateInfo_ERT000002 ,MOL_TYPE_ERT000002 );
+        
+        // Test with sample alias "test_custom"
+        executeEntryProcessTestWithSample("test_custom",templateInfo_ERT000002 ,MOL_TYPE_ERT000002 );
+
+        // Test with valid organism
+        executeEntryProcessTestWithScientificName("Homo sapiens",templateInfo_ERT000002 ,MOL_TYPE_ERT000002 );
+
+        // Test with taxId
+        executeEntryProcessTestWithTaxId("9606",templateInfo_ERT000002 ,MOL_TYPE_ERT000002 );
+
+        // Test with invalid organizm
+        executeEntryProcessInvalidOrganism("JUNK",templateInfo_ERT000002 ,MOL_TYPE_ERT000002 );
+
+    }
+    
+    public void executeEntryProcessTestWithSample(String organismName, TemplateInfo templateInfo, String molType) throws Exception{
+        
+        TemplateVariables templateVariables = getTemplateVariables_ERT000002(organismName);
+        Sample sample=getSampleForTest(templateEntryProcessor,templateVariables);
+        
+        TemplateProcessorResultSet templateProcessorResultSet = templateEntryProcessor.processEntry(templateInfo, molType, templateVariables,getOptions().getProjectId());
+
+        SourceFeature sourceFeature=templateProcessorResultSet.getEntry().getPrimarySourceFeature();
+        assertTrue(templateProcessorResultSet.getValidationResult().isValid());
+        assertEquals(sourceFeature.getTaxId(),Long.valueOf(sample.getTaxId()));
+        assertEquals(sourceFeature.getScientificName(),sample.getOrganism());
+    }
+
+    public void executeEntryProcessTestWithScientificName(String sceientificName, TemplateInfo templateInfo, String molType) throws Exception{
+        
+        TemplateVariables templateVariables = getTemplateVariables_ERT000002(sceientificName);
+        TemplateProcessorResultSet templateProcessorResultSet = templateEntryProcessor.processEntry(templateInfo, molType, templateVariables,getOptions().getProjectId());
+        
+        SourceFeature sourceFeature=templateProcessorResultSet.getEntry().getPrimarySourceFeature();
+        assertTrue(templateProcessorResultSet.getValidationResult().isValid());
+        assertEquals(sourceFeature.getScientificName(),sceientificName);
+    }
+
+    public void executeEntryProcessTestWithTaxId(String taxId, TemplateInfo templateInfo, String molType) throws Exception{
+
+        TemplateVariables templateVariables = getTemplateVariables_ERT000002(taxId);
```

Comment:
- Suggestion to pass in TemplateVariables so that the function becomes fully independent of the template being tested.

Reply:
Not doing this because we might need to initialise templateVariables for each method call.